### PR TITLE
[NeMo-UX] Integrating CLI

### PR DESCRIPTION
**Pull request by marcromeyn**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: requirements/requirements.txt [hunk 1]
```diff
@@ -1,4 +1,3 @@
-fiddle
```
Comment (jstjohn) [inline requirements/requirements.txt:1]: I assume this is gone because it's pulled in by nemo_run now. I don't see how you're adding nemo_run as a dependency though. How does that happen now? I ask since we'll be making similar changes soon to bionemo.

Comment (marcromeyn) [reply requirements/requirements.txt:1]: Yeah we will hold off on merging this PR until we have published the pypi package of nemo-run. When that's done it will replace fiddle in the requirements (nemo-run pulls in fiddle).

File: examples/llm/pretrain/img/llama3_70b_128nodes.png
Comment (titu1994) [inline examples/llm/pretrain/img/llama3_70b_128nodes.png:1]: @marcromeyn avoid putting pictures inside git - we use github releases as storage for it and link its URL where needed. See nemo docs. Fyi @ericharper

File: nemo/collections/llm/recipes/mistral.py [hunk 2]
```diff
@@ -1,61 +1,132 @@
...
-    return nl.Trainer(
-        devices=devices,
-        max_steps=100,
+    trainer = run.Config(
+        nl.Trainer,
         accelerator="gpu",
+        accumulate_grad_batches=1,
+        callbacks=callbacks,
+        devices=num_gpus_per_node,
+        gradient_clip_val=1.0,
```
Comment (akoumpa) [inline nemo/collections/llm/recipes/mistral.py:74]: @marcromeyn clip_grad should be passed to optimizerconfig, this will trigger an exception.

File: nemo/collections/llm/recipes/mixtral_8x7b_64k.py [hunk 3]
```diff
@@ -33,44 +50,37 @@ def pretrain_recipe(
...
+    recipe.model = model()
+    recipe.trainer = trainer(num_nodes=num_nodes, num_gpus_per_node=num_gpus_per_node)
+    recipe.data = run.Config(MockDataModule, seq_length=65536, global_batch_size=512, micro_batch_size=1)
 
-    model = mixtral_8x7b.model()
-    model.config.seq_length = 65536
-    model.config.max_position_embeddings = 65536
+    return recipe
 
-    trainer = mixtral_8x7b.trainer(
```
Comment (akoumpa) [inline nemo/collections/llm/recipes/mixtral_8x7b_64k.py:57]: @BoxiangW can you review these changes for the long recipes?

File: nemo/collections/llm/recipes/mixtral_8x7b_64k.py [hunk 4]
```diff
@@ -1,29 +1,46 @@
...
+    model_config.config.max_position_embeddings = 65536
+    return model_config
 
-    model = mixtral_8x7b.model()
-    model.config.seq_length = 65536
-    model.config.max_position_embeddings = 65536
 
-    trainer = mixtral_8x7b.trainer(
+def trainer(
+    num_nodes: int = 1,
```
Comment (BoxiangW) [inline nemo/collections/llm/recipes/mixtral_8x7b_64k.py:40]: Can we change the default `num_nodes` to TPPPCP/8? For all the long context recipes?

File: requirements/requirements.txt (hunk 1 above)
Comment (hemildesai) [reply requirements/requirements.txt:1]: Let's keep fiddle and add `nemo_run @ git+https://github.com/NVIDIA/NeMo-Run.git` separately for now. I will create a separate PR to move to PyPI later.

**Review by akoumpa — DISMISSED** [review]

**Review by akoumpa — APPROVED** [review]

**Review by pablo-garay — APPROVED** [review]
